dialects/core/core.thorin

```diff
@@ -122,26 +54,166 @@
+///     * `G`reater
+///     * `L`ess
+///     * `E`qual
+///
+/// | Subtag | Alias | Meaning                      |        
+/// |--------|-------|------------------------------|
+/// | `ugle` |   `f` | always false                 |              
+/// | `uglE` |   `e` | ordered and equal            |    
+/// | `ugLe` |   `l` | ordered and less             |    
+/// | `ugLE` |  `le` | ordered and less or equal    |    
+/// | `uGle` |   `g` | ordered and greater          |    
+/// | `uGlE` |  `ge` | ordered and greater or equal |    
+/// | `uGLe` |  `ne` | ordered and not equal        |    
+/// | `uGLE` |   `o` | ordered (no NaNs)            |    
+/// | `Ugle` |   `u` | unordered (either NaNs)      |    
+/// | `UglE` |  `ue` | unordered or equal           |    
+/// | `UgLe` |  `ul` | unordered or less            |    
+/// | `UgLE` | `ule` | unordered or less or equal   |    
+/// | `UGle` |  `ug` | unordered or greater         |    
+/// | `UGlE` | `uge` | unordered or greater or equa |    
+/// | `UGLe` | `une` | unordered or not equal       |    
+/// | `UGLE` |   `t` | always true                  |    
+.ax %core.rcmp(ugle =   f, uglE =   e, ugLe =   l, ugLE =  le,
+               uGle =   g, uGlE =  ge, uGLe =  ne, uGLE =   o,
+               Ugle =   u, UglE =  ue, UgLe =  ul, UgLE = ule,
+               UGle =  ug, UGlE = uge, UGLe = une, UGLE =   t):
+    Π [m: .Nat, w: .Nat] -> [%core.Real w, %core.Real w] -> .Bool, normalize_rcmp;
+///
+/// ## Conversions
+///
 /// ### %core.conv
 ///
-/// Bit width (and signedness) conversion for `.Idx -> .Idx` and `%Real -> %Real` as well as `%Real -> .Idx` and `.Idx -> %Real` conversions.
+/// Bit width (and signedness) conversion for:
+///     * `s2s`/`u2u`: `.Idx -> .Idx` 
+///     * `s2r`/`u2r`: `.Idx -> %core.Real`
+///     * `r2s`/`r2u`: `%core.Real -> .Idx` 
+///     * `r2r`: `%core.Real -> %core.Real` 
 .ax %core.conv(s2s, u2u): Π [dw: .Nat, sw: .Nat] -> .Idx sw -> .Idx dw, normalize_conv;
-.ax %core.conv(s2r, u2r): Π [dw: .Nat, sw: .Nat] -> .Idx sw -> %Real dw, normalize_conv;
-.ax %core.conv(r2s, r2u): Π [dw: .Nat, sw: .Nat] -> %Real sw -> .Idx dw, normalize_conv;
-.ax %core.conv(r2r): Π [dw: .Nat, sw: .Nat] -> %Real sw -> %Real dw, normalize_conv;
+.ax %core.conv(s2r, u2r): Π [dw: .Nat, sw: .Nat] -> .Idx sw -> %core.Real dw, normalize_conv;
+.ax %core.conv(r2s, r2u): Π [dw: .Nat, sw: .Nat] -> %core.Real sw -> .Idx dw, normalize_conv;
+.ax %core.conv(r2r): Π [dw: .Nat, sw: .Nat] -> %core.Real sw -> %core.Real dw, normalize_conv;
 ///
 /// ### %core.bitcast
 ///
 /// Bitcast to reinterpret a value as another type.
 /// Can be used for pointer / integer conversions as well as integer / nat conversions.
 .ax %core.bitcast: Π [D: *, S: *] -> S -> D, normalize_bitcast;
+///
+/// ## Other Operations
+/// 
+/// ### %core.trait
+/// 
+/// Yields the size or align of a type.
+.ax %core.trait(size, align): * -> .Nat, normalize_trait;
```

Comment:
Just a random note... how do we implement traits for other dialects' axioms?

Reply:
Good question. XD

Maybe rewrite rules can come to rescue. We have a similar situation in `%autodiff` where dialects unknown to `%autodiff` must be able to communicate to `%autodiff` that it has a really cool derivative for `%foo.bar` available:

```
.rule %autodiff.grad %foo.bar -> %foo.d_bar
```
And back to `%core.trait`:
```
.rule %core.trait %foo.Bar -> 42
```

Reply:
For now (and more generally), the dialect E that should be extended should probably make a map available and an extension function (similar to optimizations).
Then dialect A (or dialect C that depends on A and E) can register things there.